plugins/cfu/fu-cfu-device.c

```diff
@@ -404,19 +431,28 @@ fu_cfu_device_setup(FuDevice *device, GError **error)
 					      fu_hid_device_get_ep_addr_in(FU_HID_DEVICE(device)));
 	}
 
-	descriptor = fu_hid_device_parse_descriptor(FU_HID_DEVICE(device), error);
-	if (descriptor == NULL)
-		return FALSE;
-	if (!fu_cfu_device_ensure_map_item(descriptor, &self->version_get_report, error))
-		return FALSE;
-	if (!fu_cfu_device_ensure_map_item(descriptor, &self->offer_set_report, error))
-		return FALSE;
-	if (!fu_cfu_device_ensure_map_item(descriptor, &self->offer_get_report, error))
-		return FALSE;
-	if (!fu_cfu_device_ensure_map_item(descriptor, &self->content_set_report, error))
+	/* try to parse each HID descriptor -- trying to find a CFU section */
+	descriptors = fu_hid_device_parse_descriptors(FU_HID_DEVICE(device), error);
+	if (descriptors == NULL)
 		return FALSE;
-	if (!fu_cfu_device_ensure_map_item(descriptor, &self->content_get_report, error))
+	for (guint i = 0; i < descriptors->len; i++) {
+		FuHidDescriptor *descriptor = g_ptr_array_index(descriptors, i);
+		g_autoptr(GError) error_local = NULL;
+		if (!fu_cfu_device_verify_descriptor(self, descriptor, &error_local)) {
+			if (descriptors_error->len > 0)
+				g_string_append(descriptors_error, ": ");
```

Comment:
Might be worth mentioning in the message which descriptor index has the errors to make debugging life easier in future

Reply:
Good idea, fixed.